### PR TITLE
Block proof calculation is now with the correct units and is done properly from the ConsensusData (at least for PoW)

### PR DESCRIPTION
**Pull request by TheQuantumPhysicist**

### CODE REVIEW

**Review by iljakuklic — APPROVED** [review]
Nice to see these `unwrap`s go.

File: common/src/chain/block/consensus_data.rs [hunk 1]
```diff
@@ -105,4 +114,15 @@ impl PoWData {
     pub fn update_nonce(&mut self, nonce: u128) {
         self.nonce = nonce;
     }
+
+    pub fn get_block_proof(&self) -> Option<Uint256> {
```
Comment (altonen) [inline common/src/chain/block/consensus_data.rs:118]: should these be without the `get_` prefix?

File: common/src/uint/impls.rs [hunk 2]
```diff
@@ -505,6 +521,30 @@ macro_rules! construct_uint {
 construct_uint!(Uint256, 4);
 construct_uint!(Uint128, 2);
 
+impl Encode for Uint256 {
+    fn size_hint(&self) -> usize {
+        32
+    }
+
+    fn encode_to<T: parity_scale_codec::Output + ?Sized>(&self, dest: &mut T) {
```
Comment (altonen) [inline common/src/uint/impls.rs:529]: `serialization`?

File: common/src/uint/impls.rs [hunk 3]
```diff
@@ -505,6 +521,30 @@ macro_rules! construct_uint {
...
+        v.encode_to(dest)
+    }
+
+    fn encoded_size(&self) -> usize {
+        32
+    }
+}
+
+impl Decode for Uint256 {
+    fn decode<I: parity_scale_codec::Input>(
```
Comment (altonen) [inline common/src/uint/impls.rs:540]: same comment here

**Review by altonen — APPROVED** [review]

File: common/src/chain/block/consensus_data.rs (hunk 1 above)
Comment (TheQuantumPhysicist) [reply common/src/chain/block/consensus_data.rs:118]: get_ is OK if there's a calculation involved underneath.

**Pull request merged**